File: Algorithms/Medium/238_ProductOfArrayExceptSelf/Solution.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int size = nums.length;
        int[] ans = new int[size];
        int product = 1;
        
        for (int i = 0; i < size; i++) {
            ans[i] = product;
            product *= nums[i];
        }
        
        product = 1;
        for (int i = size - 1; i >= 0; i--) {
            ans[i] *= product;
            product *= nums[i];
        }
        
        
        return ans;
    }
}

File: Algorithms/Medium/238_ProductOfArrayExceptSelf/Solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix_prod = 1
        postfix_prod = 1
        res = []

        for i in range(len(nums)):
            res.append(prefix_prod)
            prefix_prod *= nums[i]
        
        for i in range(len(nums) - 1, -1, -1):
            res[i] = res[i] * postfix_prod
            postfix_prod *= nums[i]
        
        return res